### PR TITLE
#163588288 Allow access to articles even for non-authenticated users

### PR DESCRIPTION
**Pull request by Opio-Emmanuel-Omona**
#### What does this PR do?
This PR enables any user to read articles, comments and view likes but only authenticated users can create them.

#### Description of Task to be completed?
Authenticated users should be able to read as well as create the articles likes and comments
Unauthenticated users should only be able to read and when they try to create, they should be prompted to login (provide authentication credentials)

#### How should this be manually tested?
run the following commands
`git clone https://github.com/andela/Ah-backend-guardians.git`
`cd Ah-backend-guardians`
`source venv/bin/activate`
`python manage.py runserver`

Open POSTMAN and test the following endpoints both with authentication and without.
```
GET/POST 127.0.0.1:8000/api/articles/
```
```
GET/POST 127.0.0.1:8000/api/<slug>/comments/
```
```
GET/PUT 127.0.0.1:8000/api/<slug>/comments/<id>/likes/
```

The POST article data should be in the form
```
{
	"title": "Hello World",
	"body": "print('Hello world')",
	"description": "My first program"
}
```
The POST comment data should be in the form
```
{
	"body": "Cool now try it with Java"
}
```

Expected Responses For Unauthenticated Users
All POSTS and PUTS should be in the form
```
{
    "detail": "Authentication credentials were not provided."
}
```
All the GETS should provide the data

Expected Responses For Authenticated Users should be as was before.

#### What are the relevant pivotal tracker stories?
#163588288

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -64,9 +64,10 @@ def retrieve(self, request, slug):
             article = Article.objects.get(slug=slug)
             serializer = self.serializer_class(
                 article, context={'request': request})
-            if article.author.pk != request.user.pk:
-                ReadingStat.objects.get_or_create(
-                    user=request.user, articles=article)
+            if request.user.username:
```
Comment (VivianDoreen) [inline authors/apps/articles/views.py]: Avoid unnecessarily nesting if statements

Comment (Opio-Emmanuel-Omona) [reply authors/apps/articles/views.py]: @VivianDoreen I have made the changes.

**Review by VivianDoreen — APPROVED** [review]
LGTM

**Review by BarnaTB — APPROVED** [review]
Great work @Opio-Emmanuel-Omona !

**Review by archbishop12 — APPROVED** [review]
LGTM @Opio-Emmanuel-Omona I am merging this.

**Pull request merged**